Спринт 10 рейтинг фильмов начало

pavelrk97 opened the pull request.

Commit:
Спринт 10 рейтинг фильмов начало

NikolaiDem commented on src/main/java/ru/yandex/practicum/filmorate/controller/FilmController.java:
```diff
@@ -1,7 +1,61 @@
+            return newFilm;
+        } else {
+            log.error("Фильм с id = {} не найден", id);
+            throw new ValidationException("Фильм с id = " + id + " не найден");
```
Лучше здесь бросить NotFoundException, которое обработать в отдельном ExceptionHandler-e с возвратом кода ответа 404. А ещё нужно обработать исключения, генерируемые по проверке Valid, и по ним лучше вернуть 400

Commit:
Спринт 10 рейтинг фильмов
Исправлена валидация, применен @ControllerAdvice

Commit:
Спринт 10 рейтинг фильмов
Исправлена валидация, применен @ControllerAdvice

Commit:
Спринт 10 рейтинг фильмов
Исправлена валидация, применен @ControllerAdvice (мб неумело)
--== Было все хорошо, но не пускали тесты, возврата json, для этого сделал класс, чтобы String в json и после его отправка ==--
